engine/src/chunk.cpp

```diff
@@ -2283,6 +2283,191 @@ static void skip_word(const char *&sptr, const char *&eptr)
+                                   MCRangeMakeMinMax(x_mark . start, x_mark . finish),
+                                   p_delimiter,
+                                   MCRangeMakeMinMax((uindex_t)(p_first - 1), (uindex_t)p_last),
+                                   ctxt . GetStringComparisonType(),
+                                   t_range);
+    
+    x_mark . start = t_range . offset;
+    x_mark . finish = t_range . offset + t_range . length;
+}
+
+static inline void __MCCRefMarkLineRangeForEval(MCExecContext& ctxt, int4 p_first, int4 p_last, MCMarkedText& x_mark)
+{
+    __MCCRefMarkDelimitedChunkRangeForEval(ctxt,
+                                           CT_LINE,
+                                           ctxt . GetLineDelimiter(),
+                                           p_first,
+                                           p_last,
+                                           x_mark);
+}
+
+static inline void __MCCRefMarkItemRangeForEval(MCExecContext& ctxt, int4 p_first, int4 p_last, MCMarkedText& x_mark)
+{
+    __MCCRefMarkDelimitedChunkRangeForEval(ctxt,
+                                           CT_ITEM,
+                                           ctxt . GetItemDelimiter(),
+                                           p_first,
+                                           p_last,
+                                           x_mark);
+}
+
+#define __MCCRefMarkChunkForEval(chunk, ctxt, cref, x_mark) \
+        __MCCRefMarkForEval< chunk, __MCCRefMarkChunkRangeForEval<chunk> >(ctxt, cref, x_mark)
+
+void MCChunk::mark_for_eval(MCExecContext& ctxt, MCMarkedText& x_mark)
+{
+    if (cline != nil &&
+        !__MCCRefMarkForEval<CT_LINE, __MCCRefMarkLineRangeForEval>(ctxt, cline, x_mark))
+        return;
+    
+    if (paragraph != nil &&
+        !__MCCRefMarkChunkForEval(CT_PARAGRAPH, ctxt, paragraph, x_mark))
+        return;
+    
+    if (sentence != nil &&
+        !__MCCRefMarkChunkForEval(CT_SENTENCE, ctxt, sentence, x_mark))
+        return;
+        
+    if (item != nil &&
+        !__MCCRefMarkForEval<CT_ITEM, __MCCRefMarkItemRangeForEval>(ctxt, item, x_mark))
+        return;
+        
+    if (word != nil &&
+        !__MCCRefMarkChunkForEval(CT_WORD, ctxt, word, x_mark))
+        return;
+    
+    if (trueword != nil &&
+        !__MCCRefMarkChunkForEval(CT_TRUEWORD, ctxt, trueword, x_mark))
+        return;
+        
+    if (token != nil &&
+        !__MCCRefMarkChunkForEval(CT_TOKEN, ctxt, token, x_mark))
+        return;
+        
+    if (character != nil &&
+        !__MCCRefMarkChunkForEval(CT_CHARACTER, ctxt, character, x_mark))
+        return;
+    
+    if (codepoint != nil &&
+        !__MCCRefMarkChunkForEval(CT_CODEPOINT, ctxt, codepoint, x_mark))
+        return;
+    
+    if (codeunit != nil &&
+        !__MCCRefMarkChunkForEval(CT_CODEUNIT, ctxt, codeunit, x_mark))
+        return;
+        
+    if (byte != nil &&
+        !__MCCRefMarkChunkForEval(CT_BYTE, ctxt, byte, x_mark))
+        return;
+}
```

Comment:
Please add an fallback (e.g. `MCUnreachable()`) for if execution falls through all these tests.